macros/sql_gen/gen_joined_metrics_cte.sql

```diff
@@ -139,11 +139,21 @@
     {%- else %}
         join_metrics__{{previous_cte_number}}.*
     {%- endif %}
-    {%- for metric in metric_tree.ordered_expression_set %}
+    {%- for metric in metric_tree.expression_set %}
         {%- if metric_tree.ordered_expression_set[metric] == cte_number %}
-        ,({{metrics_dictionary[metric].expression | replace(".metric_value","")}}) as {{metrics_dictionary[metric].name}}
+            {%- set metric_definition = metrics_dictionary[metric].expression %}
+            {%- if "/" in metric_definition -%}
+                {%- set split_division_metric = metric_definition.split('/') -%}
+                {%- set dividend = split_division_metric[0] -%}
+                {%- set divisors = split_division_metric[1:] | list -%}
+                {%- set expression = dividend ~ " / nullif(" ~ divisors | join(", 0) / nullif(") ~ ", 0)" -%}
+            {%- else -%}
+                {%- set expression = metric_definition -%}
+            {%- endif %}
+        , ({{ expression | replace(".metric_value","") }}) as {{ metrics_dictionary[metric].name }}
```

Comment:
Where is `.metric_value` coming from in the expression that it can be replaced with nothing? It looks like this is the expression field from the metric, but that's going to be something like `{{ metric('abc') }} + {{ metric('def') }}`, right? so where is what looks to be a column of a table coming from?

Reply:
this was here before I got in here tbh! not sure what the goal here is!

Reply:
This is a holdover from the original way that we compiled down the metric! In 1.2 the expression of `{{ metric('abc') }} + {{ metric('def') }}` would have compiled to `abc.metric_value + def.metric_value`. We've resolved that in newer versions **but** its one of those things that I've kept around because it doesn't hurt. If I had to put a deprecation date on it, I'd probably go with 1.4 release of core.

Reply:
> In 1.2

so does this mean that the currently shipping version of the package, which requires 1.3 or higher, doesn't have this behaviour? In that case couldn't we remove it now instead of waiting for 1.4?

Reply:
We can punt on this for a different PR.